### ISSUE TITLE
Error in error reporting

### ISSUE THREAD

**Issue by SolalPirelli** [Zeta611/simple-bnf#4]
Thanks for the package, it looks useful!

However, with `pdfTeX, Version 3.141592653-2.6-1.40.24 (MiKTeX 22.8.28)`, I get `Unknown message 'Couldnotparser' for module 'simplebnf'`, which looking at the source code seems to be in [a line that is intended to report an error](https://github.com/Zeta611/simplebnf/blob/master/simplebnf.sty#L188). So I assume I did something wrong, but there's an error in the error-reporting code?

**Comment by Zeta611** [comment]
Hi, can you show me the grammar you wrote in the grammar environment?

**Comment by SolalPirelli** [comment]
Weirdly enough, it fails even with an empty bnfgrammar, or a trivial one (using `texify --pdf test.tex`):

```
\documentclass[acmsmall]{acmart}
\usepackage{simplebnf}
\begin{document}
Hello
\begin{bnfgrammar}
a ::= b;;
\end{bnfgrammar}
\end{document}                  
```

However, if I comment out the `bnfgrammar` block, then compile, then uncomment it, the grammar does get rendered, even though I get the `Unknown message 'Couldnotparser' for module 'simplebnf'` error. However, bibtex can't find my references any more, unless I first comment out bnfgrammar, compile with the newly added cites, then uncomment bnfgrammar and compile again.

Basically, the document can't be compiled from scratch, but it works if I first remove bnfgrammar, compile, then add it back.

I thought I had made a mistake in the grammar at some point but since even an empty bnfgrammar block fails, I guess there's something wrong somewhere.

**Comment by Zeta611** [comment]
I currently don't have a Windows machine to reproduce the problem, but I suspect the trailing `;;` to be a problem.
It should be used only as a delimiter between grammars.
Can you remove it and test if it works as you intended?

**Comment by SolalPirelli** [comment]
I checked, it happens both on Windows and on Linux for me. And I get this weird meta-error even if the bnfgrammar is empty:

```
\documentclass[acmsmall]{acmart}
\usepackage{simplebnf}
\begin{document}
Hello
\begin{bnfgrammar}
\end{bnfgrammar}
\end{document}   
```

Perhaps the current code makes some assumptions about the exact LaTeX version / distribution / ... ?

**Comment by Zeta611** [comment]
I'm sorry, but I cannot reproduce it on my Linux machine (Manjaro 5.15.65-1-MANJARO) or my Macs with:
```tex
% \documentclass[acmsmall]{acmart} works fine too
\documentclass{article}
\usepackage{simplebnf}
\begin{document}
Hello
\begin{bnfgrammar}
  a ::= b
\end{bnfgrammar}
\end{document}
```
I freshly installed the TeXLive distribution with [this mirror](http://mirror.navercorp.com/CTAN/systems/texlive/tlnet/):
`pdfTeX 3.141592653-2.6-1.40.24 (TeX Live 2022)`

I do get an error with `a ::= b;;` or an empty environment, but this is simply a syntax error (although there is a room for improvement: we can extend the DSL or the error message).

**Comment by Zeta611** [comment]
I'm closing this until futher notice.
